internal/controller/state/dataplane/configuration.go

```diff
@@ -168,6 +172,79 @@ func buildPassthroughServers(gateway *graph.Gateway) []Layer4VirtualServer {
 	return passthroughServers
 }
 
+// buildTCPServers builds TCPServers from TCPRoutes attached to listeners.
+func buildTCPServers(gateway *graph.Gateway) []Layer4VirtualServer {
+	var tcpServers []Layer4VirtualServer
+
+	for _, l := range gateway.Listeners {
+		if !l.Valid || l.Source.Protocol != v1.TCPProtocolType {
+			continue
+		}
+
+		if len(l.L4Routes) > 1 {
+			fmt.Printf(
+				"WARN: Listener %s has %d TCPRoutes, which is not supported. Skipping.",
+				l.Name,
+				len(l.L4Routes),
+			)
+			continue
+		}
```

Comment:
If this warning occurs, will the user see this information in their resource status?

Reply:
We shouldn't be using print statements in the code anyway. We have a logger to use. Regardless, we should be doing validation and setting status conditions before we even get to this point in the code. It would be when we build the graph, which is the initial step. If we're building the configuration here, this validation should have already occurred.